share/picongpu/examples/IncidentField/include/picongpu/param/incidentField.param

```diff
@@ -50,25 +51,36 @@ namespace picongpu
                 static constexpr uint32_t CENTER_Y_CELLS = 40;
                 static constexpr uint32_t WIDTH_Y_CELLS = 20;
 
-                HDINLINE float_X getTimeComponentSI(const float_X currentStep) const
+                // Get value to calculate Ey / Bz for the XMin source
+                HDINLINE float_X getXMinSourceValueSI(const floatD_X& totalCellIdx, const float_X currentStep) const
                 {
                     auto const durationStepsMiddle = DURATION_STEPS / 2;
                     auto const normalizedDuration = (currentStep - durationStepsMiddle) / durationStepsMiddle;
-                    auto const timeSI = currentStep * SI::DELTA_T_SI;
-                    const float_X sinArg = precisionCast<float_X>(timeSI / TIME_PERIOD_SI * 2.0 * PI);
-                    return AMPLITUDE_SI * math::exp(-normalizedDuration * normalizedDuration) * math::sin(sinArg);
+                    auto const k = 2.0_X * M_PI / WAVE_LENGTH_SI;
+                    auto const x = totalCellIdx.x() * SI::CELL_HEIGHT_SI;
+                    auto const t = currentStep * SI::DELTA_T_SI;
+                    // note: one could also add x influence to normalizedDuration under exp
+                    auto const longitudinal = math::exp(-normalizedDuration * normalizedDuration)

```

Comment:
Your laser definition seems odd. With the definition of `normalizedDuration` the longitudinal profile here always starts at e^{-1}, which is a very sudden jump from 0 and usually results in field solver issues with noise.

I would suggest to ask a for a `PULSE_DURATION_SI` and set an `initialization_duration = 16*PULSE_DURATION_SI`. 
Then time can be rescaled by `(current_step*DELTA_T_SI - initialization _duration_steps_half) / PULSE_DURATION_SI`.

Reply:
Oh that was a really stupid setup then. Thanks for pointing out!

Reply:
Done, the artifacts indeed disappeared.